packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -1014,17 +1014,24 @@ class _AdbLogReader extends DeviceLogReader {
   @override
   String get name => device.name;
 
-  void _start() {
+  Future<void> _start() async {
     final String lastTimestamp = device.lastLogcatTimestamp;
     // Start the adb logcat process and filter the most recent logs since `lastTimestamp`.
     final List<String> args = <String>[
       'logcat',
       '-v',
       'time',
-      '-T',
-      lastTimestamp ?? '', // Empty `-T` means the timestamp of the logcat command invocation.
     ];
-    processUtils.start(device.adbCommandForDevice(args)).then<void>((Process process) {
+    // logcat -T is not supported on Android releases before Lollipop.
+    const int kLollipopVersionCode = 21;
+    final int apiVersion = int.tryParse(await device._apiVersion);
+    if (apiVersion != null && apiVersion >= kLollipopVersionCode) {
+      args.addAll(<String>[
+        '-T',
+        lastTimestamp ?? '', // Empty `-T` means the timestamp of the logcat command invocation.
+      ]);
+    }
+    await processUtils.start(device.adbCommandForDevice(args)).then<void>((Process process) {
```

Comment:
With `await`, the `.then` is a bit redundant. You can say
```dart
final Process process = await processUtils.start(...);
_process = process;
...
```

Reply:
done